### PR TITLE
Update sleep command syntax and examples

### LINKED ISSUE #179: Specify that start and end params cannot be swapped for sleep UG
This command works: `add-sleep start/2023-01-20 02:00 end/2023-01-20 08:00`
However, this command does not works: `add-sleep end/2023-01-20 08:00 start/2023-01-20 02:00`

This bug only occurs when I swapped the parameters for start and end.

However, it is stated that this should work as in the UG you did mention that parameters can be in any order.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: sRanay/ped#6

### LINKED ISSUE #183: Change find-sleep in UG to not include date/
-------------
Labels: `type.FunctionalityBug` `severity.High`
original: marklin2234/ped#6

### LINKED ISSUE #200: Find-diet gives error for correct date format, same problem with find-sleep
find-diet gives error for correct date format

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Xuan127/ped#10

### LINKED ISSUE #208: Syntax and example given is different for edit-sleep - update UG
Syntax and example given is different

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: Xuan127/ped#13

### PR DESCRIPTION
**Pull request by DaDevChia**
This PR closes all remaining UG bugs for Sleep features, and adds a PPP for DaDevChia

- Adding a note about fixed order of parameters for sleep commands Closes #179 

- Changing find-sleep in UG to not include date Closes #183 Closes #200 

- Updating syntax and example given for edit-sleep in UG Closes #208 

- Adding a note to sleep command table Closes

### CODE REVIEW

**Review by yicheng-toh — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -447,6 +447,8 @@ Edits a single calories goal if the goal exists.
 
 ## 🛌 Sleep Management
 
+Do note that that for sleep commands, the order of the parameters is fixed, and are all non optional.
```
Comment (skylee03) [inline docs/UserGuide.md:450]:
```suggestion
Do note that for sleep commands, the order of the parameters is fixed, and are all non-optional.
```

File: docs/UserGuide.md [hunk 2]
```diff
@@ -658,6 +660,8 @@ If you forget a command, you can always use the `help` command to see their synt
 
 ### Sleep Management
 
+#### Order of parameters is fixed, and are all non optional.
```
Comment (skylee03) [inline docs/UserGuide.md:663]:
```suggestion
#### Order of parameters is fixed, and are all non-optional.
```

**Review by skylee03 — CHANGES_REQUESTED** [review]
In English, "non-" is a prefix instead of a word. Thus, we should use "non-optional" or "nonoptional" instead of "non optional". The [corpus on Google Ngram](https://books.google.com/ngrams/graph?content=non+optional%2Cnonoptional%2Cnon-optional&year_start=1950&corpus=15&smoothing=3) can support this.

Also, the duplicated "that" seem to be a clerical error.

**Pull request merged**